### PR TITLE
Add Travis test matrices for pystan 2.17 and latest

### LINKED ISSUE #294: Add tests for pystan 2.17
This should be done by updating the test matrix on travis, so it will not increase build times, and local development can still be on the pystan of your choice.

### PR DESCRIPTION
**Pull request by gentlerainsky**
Fixes #294 
Add Travis test matrices for pystan 2.17 and latest

### CODE REVIEW

**Comment by ColCarroll** [comment]
Thank you! This looks great, and the tests are failing because the test setup is not written for pystan==2.17.  

Do you mind if we wait a day or two to merge, and try to fix the underlying problems? (cc @ahartikainen)

**Comment by ahartikainen** [comment]
Oh yes. We can fix the tests for these.

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,10 @@ python:
   - "3.6"
   - "3.5"
 
+env:
+  - PYSTAN_VERSION=2.17
```
Comment (ahartikainen) [inline .travis.yml]: If we could test against 2.17.1

Comment (gentlerainsky) [reply .travis.yml]: Would you like me to change?

Comment (ahartikainen) [reply .travis.yml]: Yes, you can do it after tests are fixed.

**Comment by ahartikainen** [comment]
Hi, if you can rebase with master, so we can see everything is working.

**Comment by gentlerainsky** [comment]
@ahartikainen @ColCarroll I've rebased. There is still 1 failed test remaining.

**Comment by ahartikainen** [comment]
Yes, good that we see it. Let me see how to fix it.

**Comment by ahartikainen** [comment]
Ok, now master should work.

**Comment by ahartikainen** [comment]
LGTM

**Comment by canyon289** [comment]
@gentlerainsky Congrats on your first contribution and thanks for helping out with ArviZ!

**Comment by gentlerainsky** [comment]
@canyon289 @ahartikainen Glad I can help. Thank you for the opportunity, too.

**Pull request merged**